contracts/account/ArgentAccount.cairo

```diff
@@ -213,10 +210,38 @@ func initialize{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}
 
 @external
 func upgrade{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-    implementation: felt
-) {
+    implementation: felt, calldata_len: felt, calldata: felt*
+) -> (retdata_len: felt, retdata: felt*) {
+    alloc_locals;
     ArgentModel.upgrade(implementation);
-    return ();
+
+    if (calldata_len == 0) {
+        local retdata : felt*;
```

Comment:
I think you don't need to specify that `retdata` is local (`let retdata` is sufficient). This is taken care of by the `alloc_locals` you have at the beginning of the method.